### PR TITLE
Listen to all customer address configuration

### PR DESCRIPTION
**Pull request by indykoning**

### CODE REVIEW

File: resources/views/checkout/partials/address.blade.php [hunk 1]
```diff
@@ -41,6 +60,25 @@
...
+                label="Suffix"
+                v-model="checkout.{{ $type }}_address.suffix"
+                :required="Rapidez::config('customer/address/suffix_show', 'opt') == 'req'"
+            >
+                @if(Rapidez::config('customer/address/suffix_show', false) === 'opt')
+                    <option value=""></option>
+                @endif
+                @foreach(explode(';', Rapidez::config('customer/address/suffix_options', '')) as $suffix)
+                    <option value="{{ $suffix }}">
+                        {{ __($suffix) }}
```
Comment (royduin) [inline resources/views/checkout/partials/address.blade.php:76]:
```suggestion
                        @lang($suffix)
```

File: resources/views/checkout/partials/address.blade.php [hunk 2]
```diff
@@ -16,6 +16,25 @@
...
+                label="Prefix"
+                v-model="checkout.{{ $type }}_address.prefix"
+                :required="Rapidez::config('customer/address/prefix_show', 'opt') == 'req'"
+            >
+                @if(Rapidez::config('customer/address/prefix_show', false) === 'opt')
+                    <option value=""></option>
+                @endif
+                @foreach(explode(';', Rapidez::config('customer/address/prefix_options', '')) as $prefix)
+                    <option value="{{ $prefix }}">
+                        {{ __($prefix) }}
```
Comment (royduin) [inline resources/views/checkout/partials/address.blade.php:32]:
```suggestion
                        @lang($prefix)
```

**Review by royduin — DISMISSED** [review]

File: resources/views/checkout/partials/address.blade.php [hunk 3]
```diff
@@ -16,6 +16,25 @@
...
 <div class="contents" v-if="!$root.loggedIn || !checkout.{{ $type }}_address.customer_address_id">
+    @if(Rapidez::config('customer/address/prefix_show', false) && strlen(Rapidez::config('customer/address/prefix_options', '')))
+        <div class="col-span-12">
+            <x-rapidez::select
+                name="{{ $type }}_prefix"
+                label="Prefix"
+                v-model="checkout.{{ $type }}_address.prefix"
+                :required="Rapidez::config('customer/address/prefix_show', 'opt') == 'req'"
+            >
+                @if(Rapidez::config('customer/address/prefix_show', false) === 'opt')
```
Comment (Jade-GG) [inline resources/views/checkout/partials/address.blade.php:27]: Two different defaults? 😇

File: resources/views/checkout/partials/address.blade.php [hunk 4]
```diff
@@ -41,6 +60,25 @@
...
     </div>
+    @if(Rapidez::config('customer/address/suffix_show', false) && strlen(Rapidez::config('customer/address/suffix_options', '')))
+        <div class="col-span-12">
+            <x-rapidez::select
+                name="{{ $type }}_suffix"
+                label="Suffix"
+                v-model="checkout.{{ $type }}_address.suffix"
+                :required="Rapidez::config('customer/address/suffix_show', 'opt') == 'req'"
+            >
+                @if(Rapidez::config('customer/address/suffix_show', false) === 'opt')
```
Comment (Jade-GG) [inline resources/views/checkout/partials/address.blade.php:71]: Same here ^

File: src/Http/ViewComposers/ConfigComposer.php [hunk 5]
```diff
@@ -39,17 +39,19 @@ public function compose(View $view)
...
 
-    public function getCustomerFields()
+    public function getCustomerAddressFields()
     {
         return [
+            'prefix'      => strlen(Rapidez::config('customer/address/prefix_options', '')) ? Rapidez::config('customer/address/prefix_show', 'opt') : 'opt',
             'firstname'   => 'req',
             'middlename'  => Rapidez::config('customer/address/middlename_show', 0) ? 'opt' : false,
             'lastname'    => 'req',
+            'suffix'      => strlen(Rapidez::config('customer/address/suffix_options', '')) ? Rapidez::config('customer/address/prefix_show', 'opt') : 'opt',
```
Comment (Jade-GG) [inline src/Http/ViewComposers/ConfigComposer.php:54]: should be `suffix_show` here

**Review by Jade-GG — CHANGES_REQUESTED** [review]

**Comment by Jade-GG** [comment]
Defaults are still `'opt'` in one place (required prop) and `''` in the other, which one is the correct default for these?

**Review by Jade-GG — APPROVED** [review]

**Comment by indykoning** [comment]
That is correct, in validation we would want it to be optional if it wouldn't be shown. But in showing the actual field we want to listen to all options by default

**Review by royduin — APPROVED** [review]

**Pull request merged**